spring-test-activemq/src/main/java/com/jupiter/tools/spring/test/activemq/extension/ExpectedMessagesExtension.java

```diff
@@ -0,0 +1,112 @@
+package com.jupiter.tools.spring.test.activemq.extension;
+
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jupiter.tools.spring.test.activemq.annotation.ExpectedMessages;
+import com.jupiter.tools.spring.test.core.importdata.ImportFile;
+import com.jupiter.tools.spring.test.core.importdata.JsonImport;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import org.springframework.jms.core.JmsTemplate;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+/**
+ * Created on 07.02.2019.
+ *
+ * This extension provides an ability to assert sent messages after test execution.
+ * You can declare a list of expected messages in JSON format.
+ *
+ * @author Korovin Anatoliy
+ */
+public class ExpectedMessagesExtension implements BeforeAllCallback, AfterEachCallback {
+
+    private JmsTemplate jmsTemplate;
+
+    @Override
+    public void afterEach(ExtensionContext context) throws Exception {
+
+        ExpectedMessages expectedMessages = context.getRequiredTestMethod()
+                                                   .getAnnotation(ExpectedMessages.class);
+
+        if (expectedMessages == null) {
+            return;
+        }
+
+        Map<String, List<Map<String, Object>>> expected = new JsonImport(new ImportFile(expectedMessages.messagesFile())).read();
+
+        jmsTemplate.setReceiveTimeout(expectedMessages.timeout());
+        ObjectMapper mapper = new ObjectMapper();
+
+        long startTime = System.currentTimeMillis();
+        boolean processing = true;
+        while (processing) {
+            Object obj = jmsTemplate.receiveAndConvert(expectedMessages.queue());
+            if (obj == null) {
+                Assertions.fail("expected but not found: \n" + mapper.writeValueAsString(expected));
+            }
+
+            String className = obj.getClass().getCanonicalName();
+
+            if (!expected.containsKey(className)) {
+                if (expectedMessages.ignoreUnexpected()) {
+                    continue;
+                }
+                Assertions.fail("not expected but found: \n" + mapper.writeValueAsString(obj));
+            }
+
+            Map<String, Object> map = mapper.convertValue(obj, Map.class);
+            if (!expected.get(className).contains(map)) {
+                if (expectedMessages.ignoreUnexpected()) {
+                    continue;
+                }
+                Assertions.fail("not expected but found: \n" +
+                                "`" + className + "` :\n" +
+                                mapper.writeValueAsString(obj));
+            }
```

Comment:
it would be good to polish the code of this cycle, 
you can split it to pair different methods and encapsulate logic there..

Reply:
fixed